task/cache.py

```diff
@@ -64,7 +66,7 @@ def prune(self):
             pass
 
     # Read a resource from the cache or return None
-    def read(self, resource):
+    def read(self, resource: str) -> Optional[Dict[Any, Any]]:
```

Comment:
Insofar as we can be sure that this is a dict at all (and not some other valid JSON object, like `None`, a number, or a list) then we can at least be pretty sure that the key type is `str`.